Bercea_Bogdan/Week_2/Domain/Domain/CreateEmployeeOp/CreateEmployeeCmd.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Domain.Models;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Domain.Domain.CreateEmployeeOp
+{
+    public struct CreateEmployeeCmd
+    {
+        public string FirstName { get; }
+        public string LastName { get; }
+        public int Age { get; }
+        public string Address { get; }
+        public decimal Salary { get; }
+        public Gender Gender { get; }
+        public Position Position { get; }
+
+        public Restaurant Restaurant { get; }
+
+        public CreateEmployeeCmd(string firstName, string lastName, int age, string address,
+            decimal salary, Gender gender, Position position, Restaurant restaurant)
+        {
+            FirstName = firstName;
+            LastName = lastName;
+            Age = age;
+            Address = address;
+            Salary = salary;
+            Gender = gender;
+            Position = position;
+            Restaurant = restaurant;
+        }
```

Comment:
We've defined a command as being the input for an operation. Any input have an intrinsic property that allows any observer to tell if that given input is valid or not. Think of it this way, you want to implement a BuyBread operation which takes in a cash money as input. 

Whoever executes that operations (in a real life scenario, a vendor), should be able to observe that intrinsic property by telling if the request is valid or not by simply acknowledging if the money they received is real money or Monopoly money. [Here ](https://stackoverflow.com/questions/7926693/validation-using-attributes) you can find a possible approach to that.

Feel free to search for alternatives if that doesn't suit you well. 

Note: you can search for documentation from Microsoft as well, but note that most of their documentation revolves around asp.net models. You can tap into that, but do ignore everything web related. The stackoverflow post shows you how to invoke the validation manually.

Reply:
While I was writing that overview, I realized that that StackOverflow post is not actually the best resource. He's trying there to validate specific properties. Please check the overview for a simple example.